### PR TITLE
adding getresponseheaders method to network delegate

### PR DESCRIPTION
**Pull request by deepak1556**

### CODE REVIEW

**Comment by deepak1556** [comment]
@zcbenz the crash i was talking about was caused due to scope deletion of [network_delegate_](https://github.com/atom/brightray/blob/master/browser/url_request_context_getter.h#L68) , fixed now. No wonder it was just in my machine ;)

File: browser/network_delegate.cc [hunk 1]
```diff
@@ -53,18 +55,21 @@ int NetworkDelegate::OnHeadersReceived(
     const net::HttpResponseHeaders* original_response_headers,
     scoped_refptr<net::HttpResponseHeaders>* override_response_headers,
     GURL* allowed_unsafe_redirect_url) {
+  std::string headers;
+  original_response_headers->GetNormalizedHeaders(&headers);
+  headers_map_[request->url()] = headers;
```
Comment (zcbenz) [inline browser/network_delegate.cc]: The `headers_map_` grows infinitely.

File: browser/network_delegate.cc [hunk 2]
```diff
@@ -119,4 +124,8 @@ bool NetworkDelegate::OnCancelURLRequestWithPolicyViolatingReferrerHeader(
   return false;
 }
 
+std::string NetworkDelegate::GetResponseHeaders(const GURL& url) {
+  return headers_map_[url];
```
Comment (zcbenz) [inline browser/network_delegate.cc]: `headers_map_` is written on IO thread and read on UI thread.

**Comment by zcbenz** [comment]
Maybe we should patch `ResourceRequestDetails` to expose request's headers?

**Comment by deepak1556** [comment]
> Maybe we should patch ResourceRequestDetails to expose request's headers?

That would make it even easier to use on UI thread. Should i fix this or can work on that patch ?

**Comment by deepak1556** [comment]
can you revert this commit then https://github.com/atom/electron/commit/6f806190b62b7a0b6424aa72b218d1af2c8adaaa ? i will submit a new PR.

**Comment by zcbenz** [comment]
> That would make it even easier to use on UI thread. Should i fix this or can work on that patch ?

Patching libchromiumconent seems more decent for this.

**Comment by zcbenz** [comment]
It has been reverted at https://github.com/atom/electron/commit/4ae1998d09563dae4861e1df533c600852e3c304.

**Pull request closed without merging**